02.calendar/cal.rb

```diff
@@ -0,0 +1,56 @@
+#!/usr/bin/env ruby
+require 'date'
+require 'optparse'
+
+YEAR_MIN = 1970
+YEAR_MAX = 2100
+LINE_WIDTH = 20
+DAY_WIDTH = 2
+
+def get_input_date
+  opt = OptionParser.new
+  today = Date.today
+
+  year = today.year
+  opt.on('-y YEAR', Integer) do |input_year|
+    year = input_year if input_year.between?(YEAR_MIN, YEAR_MAX)
+  end
+
+  month = today.month
+  opt.on('-m MONTH', Integer) do |input_month|
+    month = input_month if input_month.between?(1, 12)
+  end
+
+  opt.parse!(ARGV)
+  [year, month]
+end
+
+def print_calendar(year, month)
+  puts "#{month}月 #{year}".center(LINE_WIDTH)
+  puts "日 月 火 水 木 金 土"
+
+  first_date = Date.new(year, month, 1)
+  last_date = Date.new(year, month, -1)
+  date_text = ""
+  today = Date.today
```

Comment:
```suggestion
  today = Date.new(2025, 4, 1)
```

のように、1週目にtodayがやってくるときは表示が崩れませんか？